File: 34/cpp/main.cpp
#include <algorithm>
#include <iostream>
#include <vector>

class Solution {
public:
    std::vector<int> searchRange(std::vector<int>& nums, int target) {
        // Use std::ranges::lower_bound and upper_bound
        auto first = std::ranges::lower_bound(nums, target);
        auto last = std::ranges::upper_bound(nums, target);

        // Initialize the result
        std::vector<int> result;
        result.reserve(2);

        if (first != nums.end() && *first == target) {
            result.emplace_back(std::distance(nums.begin(), first));
            result.emplace_back(std::distance(nums.begin(), last) - 1);
        } else {
            result = {-1, -1};
        }

        return result;
    }
};

int main() {
    Solution sol;

    // Test case X: Y
    RETURN_TYPE resultX = sol.FUNCTION_NAME(Y);
    std::cout << "Test X: " << resultX << std::endl;

    return 0;
}

File: 34/python3/main.py
#!/usr/bin/env python3

from bisect import bisect_left, bisect_right

class Solution:
    def searchRange(self, nums, target):
        # Find the lower bound and upper bound
        first = bisect_left(nums, target)
        last = bisect_right(nums, target) - 1

        # Check if the target is within the bounds
        if first < len(nums) and nums[first] == target:
            return [first, last]
        else:
            return [-1, -1]
